### PR TITLE
Add command to delete interview

### LINKED ISSUE #78: Delete interviews
As an user, I can delete interviews from an application, so that I can remove those that are no longer relevant.

### PR DESCRIPTION
**Pull request by czhiruo**
Users can now delete an existing interview from an existing job

### CODE REVIEW

File: src/main/java/seedu/application/logic/commands/InterviewDeleteCommand.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    private final Index interviewIndex;
+
+    /**
+     * Creates an InterviewDeleteCommand with the specified job and interview
+     *
+     * @param jobIndex The index of the job to be deleted.
+     * @param interviewIndex The index of the interview to be deleted.
+     */
+    public InterviewDeleteCommand(Index jobIndex, Index interviewIndex) {
+        requireNonNull(jobIndex);
```
Comment (migfoo02) [inline src/main/java/seedu/application/logic/commands/InterviewDeleteCommand.java:39]: Can consider just using `requireAllNonNull(jobIndex, interviewIndex)` here

File: src/main/java/seedu/application/logic/parser/CliSyntax.java [hunk 2]
```diff
@@ -15,5 +15,6 @@ public class CliSyntax {
     public static final Prefix PREFIX_INTERVIEW_TYPE = new Prefix("it/");
     public static final Prefix PREFIX_INTERVIEW_ADDRESS = new Prefix("ia/");
     public static final Prefix PREFIX_INTERVIEW_DATETIME = new Prefix("id/");
+    public static final Prefix PREFIX_INTERVIEW_DELETE = new Prefix("from/");
```
Comment (migfoo02) [inline src/main/java/seedu/application/logic/parser/CliSyntax.java:18]: Can change to `PREFIX_INTERVIEW_DELETE_SOURCE` to make it clearer

**Review by migfoo02 — APPROVED** [review]
Minor changes to improve readability.

File: src/main/java/seedu/application/logic/parser/InterviewCommandParser.java [hunk 3]
```diff
@@ -19,45 +25,62 @@
...
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
         }
 
-        switch (subCommandWord) {
+        final String commandWord = matcher.group("commandWord");
+        final String arguments = matcher.group("arguments");
+
+        // Note to developers: Change the log level in config.json to enable lower level (i.e., FINE, FINER and lower)
+        // log messages such as the one below.
+        // Lower level log messages are used sparingly to minimize noise in the code.
```
Comment (JeremyYong128) [inline src/main/java/seedu/application/logic/parser/InterviewCommandParser.java:51]: Good use of logging here!

**Review by JeremyYong128 — APPROVED** [review]
LGTM

**Pull request merged**